wremnants/muon_efficiencies.py

```diff
@@ -66,52 +66,90 @@ def make_muon_efficiency_helpers(filename = data_dir + "/testMuonSF/scaleFactorP
 
-    sf_tracking_reco = None
+    sf_tracking = None
+    sf_reco = None
 
     # tracking and reco SFs
     for nom_syst, nom_syst_tag in nom_systs.items():
-        hist_name = f"fullSF2D_{nom_syst_tag}_trackingReco_{eratag}"
+        ## first tracking
+        hist_name = f"fullSF2D_{nom_syst_tag}_tracking_{eratag}"
         hist_root = fin.Get(hist_name)
         hist_hist = narf.root_to_hist(hist_root, axis_names = ["SF eta", "SF pt"])
 
-        # pt has only one bin, so project it out for the moment
-        hist_hist = hist_hist.project("SF eta")
+        if sf_tracking is None:
+            axis_eta_eff_tracking = hist_hist.axes[0]
+            axis_pt_eff_tracking = hist_hist.axes[1]
+            sf_tracking = hist.Hist(axis_eta_eff_tracking, axis_pt_eff_tracking, axis_charge, axis_nom_alt,  name = "sf_tracking", storage = hist.storage.Weight())
 
-        if sf_tracking_reco is None:
-            axis_eta_eff_tracking_reco = hist_hist.axes[0]
-            sf_tracking_reco = hist.Hist(axis_eta_eff_tracking_reco, axis_charge, axis_nom_alt,  name = "sf_tracking_reco", storage = hist.storage.Weight())
+        # same SFs for + and - for now
+        sf_tracking.view(flow=True)[:, :, axis_charge.index(-1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
+        sf_tracking.view(flow=True)[:, :, axis_charge.index(1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
+
+        ## FIXME: avoid code duplication?
+        ## now again for reco
+        hist_name = f"fullSF2D_{nom_syst_tag}_reco_{eratag}"
+        hist_root = fin.Get(hist_name)
+        hist_hist = narf.root_to_hist(hist_root, axis_names = ["SF eta", "SF pt"])
 
+        if sf_reco is None:
+            axis_eta_eff_reco = hist_hist.axes[0]
+            axis_pt_eff_reco = hist_hist.axes[1]
+            sf_reco = hist.Hist(axis_eta_eff_reco, axis_pt_eff_reco, axis_charge, axis_nom_alt,  name = "sf_reco", storage = hist.storage.Weight())
 
         # same SFs for + and - for now
-        sf_tracking_reco.view(flow=True)[:, axis_charge.index(-1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
-        sf_tracking_reco.view(flow=True)[:, axis_charge.index(1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
+        sf_reco.view(flow=True)[:, :, axis_charge.index(-1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
+        sf_reco.view(flow=True)[:, :, axis_charge.index(1.), axis_nom_alt.index(nom_syst)] = hist_hist.view(flow=True)[...]
 
+        
     # set overflow and underflow equal to adjacent bins
-    sf_tracking_reco.view(flow=True)[0, ...] = sf_tracking_reco.view(flow=True)[1, ...]
-    sf_tracking_reco.view(flow=True)[axis_eta_eff_tracking_reco.extent-1, ...] = sf_tracking_reco.view(flow=True)[axis_eta_eff_tracking_reco.extent-2, ...]
+    sf_tracking.view(flow=True)[0, ...] = sf_tracking.view(flow=True)[1, ...]
```

Comment:
You don't have to use .view here, and I think it would be a bit clearer to use hist.overflow/hist.underflow, as done here: https://github.com/WMass/WRemnants/blob/main/wremnants/theory_corrections.py#L115-L129

Really we should make a common function to do this (option to set either 1 or nearest value)

Reply:
Should it become the following?
Maybe you just wanted to use hist.underflow (hist.overflow) instead of 0 (axis.extent-1) in the line above?

if sf_tracking.axes[0].traits.underflow:
        sf_tracking[hist.underflow,...] = sf_tracking[1,...].view(flow=True)
    if sf_tracking.axes[0].traits.overflow:
        sf_tracking[hist.overflow,...] = sf_tracking[sf_tracking.axes[0].extent-2,...].view(flow=True)

Reply:
It would be sf_tracking[hist.underflow,...] = sf_tracking[0,...].view(flow=True),

or sf_tracking[hist.underflow,...] = sf_tracking.view(flow=True)[1,...], since the offset by one only happens when you do view(flow=True), which returns an array with the underflow included I think the former is a bit clearer

For the last entry you can do 

sf_tracking[hist.overflow,...] = sf_tracking[-1,...].view(flow=True)

Reply:
Done, but please check it is correct now